component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/reflect/ParameterModelService.java

```diff
@@ -227,9 +232,18 @@ private Map<String, String> buildExtensions(final String name, final Type generi
     }
 
     private Stream<Annotation> getAnnotations(final Type type, final Annotation[] annotations) {
+        // we have a few implicit constraints. those constraints can be overwritten by explicit one.
+        // to avoid ambiguous exception we should filter out implicit annotation constraints
+        final Set<Class<? extends Annotation>> uniqueAnnotations = new HashSet<>(Arrays.asList(Min.class, Max.class));
+        final Set<? extends Class<? extends Annotation>> forbidImplicit = Stream.of(annotations)
+                .map(Annotation::annotationType)
+                .filter(uniqueAnnotations::contains)
+                .collect(toSet());
         return Stream
                 .concat(getReflectionAnnotations(type, annotations),
-                        implicitAnnotationsMapping.getOrDefault(type, emptyList()).stream());
+                        implicitAnnotationsMapping.getOrDefault(type, emptyList())
+                                .stream()
+                                .filter(it -> !forbidImplicit.contains(it.annotationType())));
```

Comment:
Could it be optimized to use uniqueAnnotations instead of a separate forbidImplicit:
.filter(it -> !uniqueAnnotations.contains(it.annotationType())))

Reply:
It won't be the same. The variable name I may rename.